interfaces/builtin/io_ports_control_test.go

```diff
@@ -25,70 +25,104 @@ import (
 	"github.com/snapcore/snapd/interfaces"
 	"github.com/snapcore/snapd/interfaces/builtin"
 	"github.com/snapcore/snapd/snap"
-	"github.com/snapcore/snapd/testutil"
+	"github.com/snapcore/snapd/snap/snaptest"
 )
 
-type IoPortsControlInterfaceSuite struct {
+type IioPortsControlInterfaceSuite struct {
 	iface interfaces.Interface
 	slot  *interfaces.Slot
 	plug  *interfaces.Plug
 }
 
-var _ = Suite(&IoPortsControlInterfaceSuite{
-	iface: builtin.NewIoPortsControlInterface(),
-	slot: &interfaces.Slot{
-		SlotInfo: &snap.SlotInfo{
-			Snap:      &snap.Info{SuggestedName: "core", Type: snap.TypeOS},
-			Name:      "io-ports-control",
-			Interface: "io-ports-control",
-		},
-	},
-	plug: &interfaces.Plug{
-		PlugInfo: &snap.PlugInfo{
-			Snap:      &snap.Info{SuggestedName: "other"},
-			Name:      "io-ports-control",
-			Interface: "io-ports-control",
-		},
-	},
+var _ = Suite(&IioPortsControlInterfaceSuite{
+	iface: &builtin.IioPortsControlInterface{},
 })
 
-func (s *IoPortsControlInterfaceSuite) TestName(c *C) {
+func (s *IioPortsControlInterfaceSuite) SetUpTest(c *C) {
+	// Mock for OS Snap
+	osSnapInfo := snaptest.MockInfo(c, `
+name: ubuntu-core
+type: os
+slots:
+  test-io-ports:
+    interface: io-ports-control
+`, nil)
+	s.slot = &interfaces.Slot{SlotInfo: osSnapInfo.Slots["test-io-ports"]}
+
+	// Snap Consumers
+	consumingSnapInfo := snaptest.MockInfo(c, `
+name: client-snap
+plugs:
+  plug-for-io-ports:
+    interface: io-ports-control
```

Comment:
@tonyespy - this issues with this section have to do with how the mocking is being done. Not enough is being mocked to do it in the way I described and there aren't any examples to mock an implicit slot. I suggest for clarity of intent in the test to simply change consumingSnapInfo to be:
```
name: client-snap                                                               
plugs:                                                                          
  plug-for-io-ports:                                                            
    interface: io-ports-control                                                 
apps:                                                                           
  app-accessing-io-ports:                                                       
    command: foo                                                                
    plugs: [plug-for-io-ports] 
```

In this manner you are plugging what you specified in toplevel 'plugs', which is how a developer would normally do this. Alternatively, you could revamp the setup to work like 'network-control'. That's harder and busy work at this point, so I suggest what I outlined above.